src/Sentry/Scope.cs

```diff
@@ -225,6 +225,20 @@ public ITransactionTracer? Transaction
             try
             {
                 _transaction.Value = value;
+
+                if (Options.EnableScopeSync)
+                {
+                    if (_transaction.Value != null)
+                    {
+                        // If there is a transaction set we propagate the trace to the native layer
+                        Options.ScopeObserver?.SetTrace(_transaction.Value.TraceId, _transaction.Value.SpanId);
```

Comment:
Is it just the TraceId and SpanId that need to be synced? Should the sample rate and sample rand also be synced (we normally put that on the DSC to ensure consistent sampling decisions)? Or will this just be used to create spans (not transactions) and so no sampling decisions need to be made?

Reply:
In the future, they should. But both the iOS and Android ScopeObservers are missing their implementation. See https://github.com/getsentry/sentry-dotnet/issues/4074

It needs bumping the native SDKs to specific versions and then mess around with the bindings (especially, since the API is in private/internal parts of the Cocoa/Java SDK)

Reply:
Please raise a ticket in the relevant repos so we don't forget about this